### PR TITLE
PLT-6080 moving clustering to memberlist

### PR DESCRIPTION
**Pull request by coreyhulen**
** NEEDS TO BE LOAD TESTED **

#### Summary
- Change to use hashicorp/memberlist for cluster node discovery
- Move the seeding of cluster nodes into the DB
- Added the ability to wait for cluster sending

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-5282
https://mattermost.atlassian.net/browse/PLT-6078
https://mattermost.atlassian.net/browse/PLT-6080

#### Checklist
- [X] Added or updated unit tests (required for all new features)
- [ ] Added API documentation (required for all new APIs)
- [X] All new/modified APIs include changes to the drivers
- [X] Has enterprise changes (https://github.com/mattermost/enterprise/pull/142)
- [X] Has UI changes
- [X] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates
- [X] Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

**Comment by jwilander** [comment]
A unit test is failing on a panic though

File: Makefile [hunk 1]
```diff
@@ -58,6 +58,7 @@ TESTFLAGS ?= -short
 TE_PACKAGES=$(shell go list ./... | grep -v vendor)
 TE_PACKAGES_COMMA=$(shell echo $(TE_PACKAGES) | tr ' ' ',')
 
+#EE_PACKAGES=github.com/mattermost/platform/enterprise/cluster
```
Comment (crspeller) [inline Makefile]: Remove?

File: model/cluster_discovery.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	CDS_OFFLINE_AFTER_MILLIS = 1000 * 60 * 30 // 30 minutes
+	CDS_TYPE_APP             = "mattermost_app"
+)
+
+type ClusterDiscovery struct {
+	Id          string `json:"id"`
+	Type        string `json:"type"`
+	ClusterName string `json:"cluster_name"`
+	Hostname    string `json:"hostname"`
+	GossipPort  int32  `json:"gossip_port"`
```
Comment (coreyhulen) [inline model/cluster_discovery.go]: This is so different servers can have different ports if needed.  Although we don't use this for Kubes, it was one of the complaints of the old system.

File: app/web_hub.go [hunk 3]
```diff
@@ -64,10 +64,11 @@ func TotalWebsocketConnections() int {
 }
 
 func HubStart() {
-	l4g.Info(utils.T("api.web_hub.start.starting.debug"), runtime.NumCPU()*2)
+	numberOfHubs := runtime.NumCPU() * 2
+	l4g.Info(utils.T("api.web_hub.start.starting.debug"), numberOfHubs)
 
 	// Total number of hubs is twice the number of CPUs.
```
Comment (coreyhulen) [inline app/web_hub.go]: hmm...where?

The line is almost identical to before.  I just pull out `runtime.NumCPU()*2`.

File: Makefile (hunk 1 above)
Comment (coreyhulen) [inline Makefile]: removed

File: app/web_hub.go (hunk 3 above)
Comment (crspeller) [inline app/web_hub.go]: Just saying it should be with the assignment not here.

File: model/cluster_discovery.go (hunk 2 above)
Comment (crspeller) [inline model/cluster_discovery.go]: Ok.

File: app/web_hub.go (hunk 3 above)
Comment (coreyhulen) [inline app/web_hub.go]: derp...moved the comment.  I thought you were talking about the log message.

**Pull request merged**